backtester/config/config-builder/main.go

```diff
@@ -0,0 +1,280 @@
+	}
+
+	var err error
+	cfg.StrategySettings.Name, err = parseStratName(quickParse(reader), strategiesToUse)
+
+	fmt.Println("Enter a nickname, it can help distinguish between different configs using the same strategy")
+	cfg.Nickname = quickParse(reader)
+	fmt.Println("Does this strategy have custom settings? y/n")
+	customSettings := quickParse(reader)
+	if strings.Contains(customSettings, "y") {
+		cfg.StrategySettings.CustomSettings = customSettingsLoop(reader)
+	}
+
+	fmt.Println("-----Exchange Settings-----")
+	addCurrency := "y"
+	for strings.Contains(addCurrency, "y") {
+		var currencySetting *config.CurrencySettings
+		currencySetting, err = addCurrencySetting(reader)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		cfg.CurrencySettings = append(cfg.CurrencySettings, *currencySetting)
+		fmt.Println("Add another exchange currency setting? y/n")
+		addCurrency = quickParse(reader)
+	}
+
+	if len(cfg.CurrencySettings) > 1 {
+		for i := range strats {
+			if strats[i].Name() == cfg.StrategySettings.Name &&
+				strats[i].SupportsSimultaneousProcessing() {
+				fmt.Println("Will this strategy use simultaneous processing? y/n")
+				yn := quickParse(reader)
+				if yn == "y" || yn == "yes" {
+					cfg.StrategySettings.SimultaneousSignalProcessing = true
+				}
+			}
+			break
+		}
+	}
+
+	fmt.Println("-----Portfolio Settings-----")
+	fmt.Println("Will there be global portfolio buy-side limits? y/n")
+	yn := quickParse(reader)
+	if yn == "y" || yn == "yes" {
+		cfg.PortfolioSettings.BuySide, err = minMaxParse("buy", reader)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+	fmt.Println("Will there be global portfolio sell-side limits? y/n")
+	yn = quickParse(reader)
+	if yn == "y" || yn == "yes" {
+		cfg.PortfolioSettings.SellSide, err = minMaxParse("sell", reader)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	fmt.Println("-----Data Settings-----")
+	fmt.Println("Will you be using \"candle\" or \"trade\" data?")
+	cfg.DataSettings.DataType = quickParse(reader)
+	fmt.Println("What time interval will you use?")
+	allCandles := gctkline.SupportedCandles
+	for i := range allCandles {
+		fmt.Printf("%v. %s", i+1, allCandles[i])
+	}
+	cfg.DataSettings.DataType = quickParse(reader)
+
+	fmt.Println("-----Statistics Settings-----")
+	fmt.Println("Enter the risk free rate. eg 0.03")
+	cfg.StatisticSettings.RiskFreeRate, err = strconv.ParseFloat(quickParse(reader), 64)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var resp []byte
+	resp, err = json.MarshalIndent(cfg, "", " ")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println("Write strat to file? If no, the output will be on screen y/n")
+	yn = quickParse(reader)
+	if yn == "y" || yn == "yes" {
+		var wd string
+		wd, err = os.Getwd()
+		wd = filepath.Join(wd, cfg.StrategySettings.Name+"-"+cfg.Nickname, ".strat")
+		fmt.Printf("Enter output file. If blank, will output to \"%v\"\n", wd)
+		path := quickParse(reader)
+		if path == "" {
+			path = wd
+		}
+		err = os.WriteFile(path, resp, 0770)
```

Comment:
build error: .\main.go:148:9: undefined: os.WriteFile

Reply:
This is not complete

Reply:
Okay!